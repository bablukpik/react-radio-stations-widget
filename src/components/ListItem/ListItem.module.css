.listItem {
  display: flex;
  justify-content: space-between;
  list-style-type: none;
  border-bottom: 2px solid #42424c;
  padding-top: 27px;
  padding-bottom: 28px;
}

.listItem:last-of-type {
  border-bottom: none;
}

.cover {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  margin-top: 44px;
  margin-bottom: 47px;
}

.coverImgWrapper {
  height: 280px;
  width: 280px;
  border: 4px solid #a2abbd;
  border-radius: 50%;
}

.coverImg {
  width: 100%;
}
