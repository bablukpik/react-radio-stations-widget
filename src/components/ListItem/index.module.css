.listItem {
  display: flex;
  justify-content: space-between;
  list-style-type: none;
  border-bottom: 2px solid #42424c;
  padding: 14.5px 0;
  cursor: pointer;
}

.listItem:last-of-type {
  border-bottom: none;
}
