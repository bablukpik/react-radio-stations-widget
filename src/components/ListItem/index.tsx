import React from 'react';
import {useSpring, animated} from 'react-spring';
import Button from '../Button';
import Image from '../Image';
import MinusIcon from '../../assets/icons/minus.png';
import PlusIcon from '../../assets/icons/plus.png';
import CoverImg from '../../assets/icons/cover-img.png';
import styles from './ListItem.module.css';

interface Props {
  isOpen: boolean;
  children: React.ReactNode;
  onClick: (e: any) => void;
  id: string;
}

const ListItem = ({ isOpen = false, children, ...props }: Props) => {
  const animateProps = useSpring({
    delay: 90,
    from: {opacity: 0},
    to: {opacity: 1},
  })
  return (
    <React.Fragment>
      {isOpen && (
        <animated.div className={styles.cover} style={animateProps} {...props}>
        <Button>
          <Image
            src={MinusIcon}
            alt="Minus Icon"
          />
        </Button>
        <div className={styles.coverImgWrapper}>
          <Image
            className={styles.coverImg}
            src={CoverImg}
            alt="Cover image"
          />
        </div>
  
        <Button>
          <Image src={PlusIcon} alt="Plus Icon" />
        </Button>
      </animated.div>
      )}
      <li className={styles.listItem} {...props}>
        {children}
      </li>
    </React.Fragment>
  )
}

export default ListItem;
