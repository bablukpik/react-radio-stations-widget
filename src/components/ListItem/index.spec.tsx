import { render } from '@testing-library/react';

import ListItem from './index';

const getProps = (newProps?: any) => ({
    isOpen: false,
    children: <></>,
    ...newProps,
});

describe('ListItem', () => {
  it('should render', () => {
    const { asFragment } = render(<ListItem {...getProps()}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the station name', () => {
    const props = {
        children: <span>Test FM</span>
    };
    const { getByText } = render(<ListItem {...getProps(props)}/>);
    expect(getByText('Test FM')).toBeInTheDocument();
  });

  it('should render the station id', () => {
    const props = {
        children: <span>1,1</span>
    };
    const { getByText } = render(<ListItem {...getProps(props)}/>);
    expect(getByText('1,1')).toBeInTheDocument();
  });

  it('renders cover image and icons', () => {
    const props = {
        isOpen: true,
    };
    const { getByAltText } = render(<ListItem {...getProps(props)}/>);
    expect(getByAltText('Minus icon')).toBeInTheDocument();
    expect(getByAltText('Plus icon')).toBeInTheDocument();
    expect(getByAltText('Cover image')).toBeInTheDocument();
  });
});
