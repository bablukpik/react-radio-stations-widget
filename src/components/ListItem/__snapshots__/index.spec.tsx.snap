// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render 1`] = `
<DocumentFragment>
  <li
    class="listItem"
  />
</DocumentFragment>
`;
